aws/resource_aws_wafregional_sql_injection_match_set_test.go

```diff
@@ -0,0 +1,318 @@
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_sql_injection_match_set.sql_injection_match_set", "sql_injection_match_tuple.3961339938.field_to_match.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_sql_injection_match_set.sql_injection_match_set", "sql_injection_match_tuple.3961339938.field_to_match.0.data", "user-agent"),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_sql_injection_match_set.sql_injection_match_set", "sql_injection_match_tuple.3961339938.field_to_match.0.type", "HEADER"),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_sql_injection_match_set.sql_injection_match_set", "sql_injection_match_tuple.3961339938.text_transformation", "NONE"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSWafRegionalSqlInjectionMatchSet_noTuples(t *testing.T) {
+	var ipset waf.SqlInjectionMatchSet
+	setName := fmt.Sprintf("sqlInjectionMatchSet-%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafRegionalSqlInjectionMatchSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSWafRegionalSqlInjectionMatchSetConfig_noTuples(setName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSWafRegionalSqlInjectionMatchSetExists("aws_wafregional_sql_injection_match_set.sql_injection_match_set", &ipset),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_sql_injection_match_set.sql_injection_match_set", "name", setName),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_sql_injection_match_set.sql_injection_match_set", "sql_injection_match_tuple.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSWafRegionalSqlInjectionMatchSetDisappears(v *waf.SqlInjectionMatchSet) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).wafregionalconn
+		region := testAccProvider.Meta().(*AWSClient).region
+
+		wr := newWafRegionalRetryer(conn, region)
+		_, err := wr.RetryWithToken(func(token *string) (interface{}, error) {
+			req := &waf.UpdateSqlInjectionMatchSetInput{
+				ChangeToken:            token,
+				SqlInjectionMatchSetId: v.SqlInjectionMatchSetId,
+			}
+
+			for _, sqlInjectionMatchTuple := range v.SqlInjectionMatchTuples {
+				sqlInjectionMatchTupleUpdate := &waf.SqlInjectionMatchSetUpdate{
+					Action: aws.String("DELETE"),
+					SqlInjectionMatchTuple: &waf.SqlInjectionMatchTuple{
+						FieldToMatch:       sqlInjectionMatchTuple.FieldToMatch,
+						TextTransformation: sqlInjectionMatchTuple.TextTransformation,
+					},
+				}
+				req.Updates = append(req.Updates, sqlInjectionMatchTupleUpdate)
+			}
+			return conn.UpdateSqlInjectionMatchSet(req)
+		})
+		if err != nil {
+			return errwrap.Wrapf("[ERROR] Error updating SqlInjectionMatchSet: {{err}}", err)
```

Comment:
Minor nitpick: `errwrap.Wrapf()` usage instead of `fmt.Errorf()` (same with below)